.github/workflows/main.yml

```diff
@@ -9,23 +9,39 @@ jobs:
     runs-on: ubuntu-latest
 
     strategy:
-      max-parallel: 4
+      max-parallel: 5
       matrix:
         python-version: [ "3.9", "3.10", "3.11", "3.12" , "3.13" ]
 
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v3
-        with:
-          python-version: ${{ matrix.python-version }}
-      - name: install
-        run: |
-          python -m pip install -U pip
-          pip install -e ".[dev]"
-      - name: lint
-        run: |
-          make lint
-      - name: test
-        run: |
-          make coverage
-          coverage report
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
+      with:
+        enable-cache: true
+        cache-dependency-glob: "**/pyproject.toml"
+    - name: Install dependencies
+      run: uv sync --all-extras
+    - name: Check linting
+      run: uv run poe lint
+    - name: Test with pytest
+      run: |
+        uv run coverage run -m pytest --benchmark-skip
```

Comment:
I would prefer to split the linting and testing into two jobs. The testing runs the matrix, the linter v3.13.

Reply:
yeah, my wasting compute resources got the best of me here; it is hard to follow. I'll separate.